src/Mod/Arch/ArchRebar.py

```diff
@@ -468,28 +468,18 @@ def execute(self,obj):
             elif round(influenceArea) < round(reqInfluenceArea):
                 FreeCAD.Console.PrintWarning("Last span is greater that end offset.\n")
             for i in range(len(spacinglist)):
-                if i == 0:
-                    barplacement = CustomSpacingPlacement(spacinglist, 1, axis, father.Placement.Rotation, obj.OffsetStart.Value, obj.OffsetEnd.Value)
-                    placementlist.append(barplacement)
-                else:
-                    barplacement = CustomSpacingPlacement(spacinglist, i+1, axis, father.Placement.Rotation, obj.OffsetStart.Value, obj.OffsetEnd.Value)
-                    placementlist.append(barplacement)
+                barplacement = CustomSpacingPlacement(spacinglist, i+1, axis, father.Placement.Rotation, obj.OffsetStart.Value, obj.OffsetEnd.Value)
+                placementlist.append(barplacement)
             obj.Amount = len(spacinglist)
             obj.Spacing = 0
         obj.PlacementList = placementlist
         for i in range(len(obj.PlacementList)):
-            if i == 0:
-                bar.Placement = obj.PlacementList[i]
-                shapes.append(bar)
-                basewire.Placement = obj.PlacementList[i]
-                self.wires.append(basewire)
-            else:
-                bar = bar.copy()
-                bar.Placement = obj.PlacementList[i]
-                shapes.append(bar)
-                w = basewire.copy()
-                w.Placement = obj.PlacementList[i]
-                self.wires.append(w)
+            bar = bar.copy()
+            bar.Placement = obj.PlacementList[i]
+            shapes.append(bar)
+            w = basewire.copy()
+            w.Placement = obj.PlacementList[i]
+            self.wires.append(w)
```

Comment:
Why can the specific logic for `i == 0` be removed?

Reply:
> Why can the specific logic for `i == 0` be removed?

bar is a Part shape. we can simply copy it for each bar that must be add. The first one is not something special.

Reply:
```suggestion
           if i != 0:
               bar = bar.copy()
               basewire = basewire.copy()
            bar.Placement = obj.PlacementList[i]
            shapes.append(bar)
            basewire.Placement = obj.PlacementList[i]
            self.wires.append(basewire)
```

I think we can use above logic. @ebrahimraeyat with your logic you are creating new `bar` and `basewire` objects  for `i=0` but objects are already there. Yes, Python has good garbage collector but here we can control object creation.